### ISSUE TITLE
Raison d'être of this project

### ISSUE THREAD

**Issue by michaelklishin** [temochka/queueing_rabbit#3]
Apologies for asking a question here, I find it less annoying than unsolicited emails.

I'm curious what's the raison d'être of this project? Not trying to suggest it doesn't solve a problem but  Hutch and Sneakers have been around for a while and many seem to enjoy the ease of use of the former and the throughput the latter can provide.

Otherwise `queueing_rabbit` seems to be fairly similar to Hutch. Why have you decided to develop your own alternative?

**Comment by temochka** [comment]
Hi Michael,

Well, this project is fairly old. It grew out of our internal gem at [Wildbit](http://wildbit.com) (using the amqp gem) that I created in July 2012, a few months before the amazing bunny 0.9.0 release (thank you for your work!). Back then there was almost no options to choose from, so we had to go with our own solution. We wanted something similar to Resque, but working on top of RabbitMQ with addition of declarative DSL to describe RabbitMQ topologies. Later in March 2013 I added bunny adapter and [open-sourced the gem](https://rubygems.org/gems/queueing_rabbit/versions) as queueing_rabbit (still before either Hutch or sneakers appeared) but never announced it properly or published any documentation until February 2014.

I never heard of Hutch until now, but I looked into the sneakers gem when it appeared. I found it too assuming and high-level to be useful for our applications. It’s probably better now, but we got used to queueing_rabbit, and can’t really advocate the migration at this point. I’m confident both Hutch and sneakers are good libraries and one of them would be my first choice if I was to start a new project these days.

Anyway, I guess you really came here because you saw a retweet of my tweet from over a year ago today (I’m puzzled how he found it too) and your real concern is fragmentation of tooling around RabbitMQ. Even though queueing_rabbit is maintained and works well for us, I don’t think it’s a good solution for the majority of users. Because of that, I would accept a PR adding disclaimer to the top of the README file that mentions Hutch and sneakers as more mainstream tools to do the job. I really have no intention to compete here. Let me know if this answers your question.
